### ISSUE TITLE
Increase feature-parity for builtin comparison operators (‘==", ‘<’, ‘<=’, ‘>=’, ‘>’)

### ISSUE THREAD

**Issue by Pydes-boop** [Attumm/redis-dict#38]
## Description

Currently im working with redis-dict quite a bit and using it as a drop in replacement for some features where an internal dictionary is used. While in general a `__cmp__` function is present for redis dict and most operator calls are correctly delegated there, the `==` does not compare the actual dict items as it should but only the id of the objects.

## Proposal

The `==` / `__eq__` operator should "compare equal if and only if they have the same (key, value) pairs (regardless of ordering)". Additionally i think it would be nice functionality that a redis-dict can be compared to a built-in dict and compare equal if they have the same pairs as well.

Otherwise the TODO for `__cmp__` could mostly be removed, as the other comparison operators (ie.  `<`, `<=`, `>=`, `>`) are delegated to `__cmp__` by default and show correct behaviour by raising a type error. Even though its not necessary for functionality, the functions could be added to the class as delegates to specify their behaviour in docstrings and make this easier to find by users.

This is all based on the standard dict behaviour specified here: https://docs.python.org/3/library/stdtypes.html#dict

## Why?

With the system im working with we sometimes have to switch how some of our data is temporarily stored during runtime to improve redundancy. For this it would be great if redis-dict compared with a built-in dict as this would make delegation more straightforward and we could also easily search for issues with concurrency by comparing internal dicts to the current redis-dict.

Let me know what you think, all in all this should be fairly straight forward to add and I would get started with implementing these and adding tests / docs right away if you agree with my proposed changes. Im open to any suggestions if there is anything you would prefer differently, or some functionality you would instead like to add to these operators that could be beneficial even if its outside of their expected behaviour as a dictionary.

**Comment by Attumm** [comment]
Yeah, that sounds great; it adheres to the initial philosophy of the project.
To have a dictionary that is storing its data in Redis instead of memory.

For issues like these, it could be easier to communicate and showcase the problem by presenting them in code and having a unittest for each case.

Secondly, please don't start out worrying about it, but:

What is also important, though, is that we think about performance. We want to reduce/keep the number of calls to the Redis server low. It could also be that the Redis server instance is running with more working memory than the client.

But that is, of course, secondary.

Then, lastly, if done, we will have to bump the version to 4.x.x. for stability.

So I'm all for it.

**Comment by Pydes-boop** [comment]
Thanks, i dont have the rights self-assign the issue but ill get working on it then.

Even though its not seperated in proper unit tests yet this is the initial code with which one can confirm and reproduce the described behaviours.
```
    def test_cmp(self):
        r = self.create_redis_dict(expire=3600)
        r['foobar1'] = 'barbar1'
        r['foobar2'] = 'barbar2'

        d = dict()
        d['foobar1'] = 'barbar1'
        d['foobar2'] = 'barbar2'
        r_fake = dict()
        r_fake['foobar1'] = 'barbar1'
        r_fake['foobar2'] = 'barbar2'

        # These currently return False even though the items are equal
        print(d == r)
        print(d is r)
        print(r is d)
        print(r == d)
        # This is True even though a direct eq between the dicts is false
        print(r.items() == list(d.items()))

        # expected bevahiour would be as follows:
        # These currently return False even though the items are equal
        print(r_fake == r)
        print(r_fake is r)
        print(r is r_fake)
        print(r == r_fake)
        # This is True even though a direct eq between the dicts is false
        print(r.items() == list(r_fake.items()))


        # This is correct bevahiour
        self.assertRaises(TypeError, r.__cmp__(d))
        with self.assertRaises(TypeError):
            r < r
            r <= r
            r >= r
            r > r
            r < d
            r <= d
            r >= d
            r > d
```

**Comment by Attumm** [comment]
@Pydes-boop 

How's it going? Still making progress?

**Comment by Pydes-boop** [comment]
Sorry for no updates so far, its going well but another project is just currently getting in the way. I should be able to update this PR next week for review.

**Comment by Attumm** [comment]
The PR has landed in v2.5.0.

Thanks
